src/response/phonon/ad.jl

```diff
@@ -0,0 +1,32 @@
+"""
+Compute the dynamical matrix at ``Γ``-point in reciprocal space of a system using automatic
+differentiation. In this particular case, it is equal to the Hessian of the system.
+"""
+function compute_dynmat_ad(basis::PlaneWaveBasis{T}; scf_kwargs...) where {T}
+    # TODO: Cannot use symmetries at all, see https://github.com/JuliaMolSim/DFTK.jl/issues/817
+    @assert isone(only(basis.model.symmetries))
+
+    model = basis.model
+    cell = (; model.lattice, model.atoms, model.positions)
+    n_atoms = length(model.positions)
+    n_dim = model.n_dim
+
+    dynamical_matrix = zeros(ComplexF64, (n_dim, n_atoms, n_dim, n_atoms))
+    for τ in 1:n_atoms
+        for γ in 1:n_dim
+            displacement = zero.(model.positions)
+            displacement[τ] = StaticArrays.setindex(displacement[τ], one(T), γ)
```

Comment:
Should we define our own setindex, and alias it to staticarrays'? I think there are other places in the code where `setindex` would be nice

Reply:
Sure